common/src/main/java/tachyon/util/network/NetworkAddressUtils.java

```diff
@@ -241,10 +337,164 @@ public static ServerSocket getSocket(final TServerSocket thriftSocket) {
    * @return the InetSocketAddress of the master
    */
   public static InetSocketAddress getMasterAddress(TachyonConf conf) {
-    String masterHostname =
-        conf.get(Constants.MASTER_HOSTNAME, getLocalHostName(conf));
+    String masterHost = conf.get(Constants.MASTER_HOSTNAME, getLocalHostName(conf));
     int masterPort = conf.getInt(Constants.MASTER_PORT, Constants.DEFAULT_MASTER_PORT);
-    return new InetSocketAddress(masterHostname, masterPort);
+    return new InetSocketAddress(masterHost, masterPort);
+  }
+
+  /**
+   * Helper method to get the {@link InetSocketAddress} connect address of the master.
+   *
+   * @param conf the configuration of Tachyon
+   * @return the master's connect address
+   */
+  public static InetSocketAddress getMasterConnectAddress(TachyonConf conf) {
+    if (conf.get(Constants.MASTER_BIND_HOST, "localhost").equals("0.0.0.0")) {
```

Comment:
1. After https://github.com/amplab/tachyon/pull/1218 has been patched, we can refactor current implementation.
2. Tachyon should always include a default binding value for each of services. Currently, "0.0.0.0" is the default value.